tests/test_engine.py

```diff
@@ -1284,6 +1285,155 @@ def flaky_function():
                     log_prints=False,
                 )
 
+    async def test_is_retriable_after_failure(
+        self, mock_anyio_sleep, prefect_client, flow_run, result_factory
+    ):
+        # the flow run must be running prior to running tasks
+        await prefect_client.set_flow_run_state(
+            flow_run_id=flow_run.id,
+            state=Running(),
+        )
+
+        # Define a task that fails once and then succeeds
+        mock = MagicMock()
+
+        # Can retry only once
+        def is_retriable(task, task_run, state):
```

Comment:
Can we do one more test for when the criteria is not met in the `retry_conditinon_fn` for a retry?